app/Http/Controllers/RulesController.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Controllers;
+
+use App\Models\City;
+use App\Models\Country;
+use App\Models\Rules;
+use App\Services\OpenAIService;
+
+class RulesController
+{
+    public function getRules($country, $city): array
```

Comment:
Typing
```suggestion
    public function getRules(string $country, string $city): array
```

Reply:
I would go with: 
```
public function getRules(Country $country, City $city)
```

and get city like that:
```
 $selectedCity = City::query()
      ->whereBelongsTo($country)
      ->where("id", $city->id)
      ->with("cityProviders", "country")
      ->firstOrFail();
      
 $rules = $selectedCity->rules()->get();     
```

Reply:
And even:
```
public function getRules(Country $country, City $city, OpenAIService $importer)
```